### PR TITLE
rbd: find_action() should sort actions first

### PR DESCRIPTION
**Pull request by idryomov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by idryomov** [comment]
jenkins test make check

File: src/tools/rbd/Shell.cc [hunk 1]
```diff
@@ -301,8 +301,14 @@ void Shell::get_command_spec(const std::vector<std::string> &arguments,
...
 Shell::Action *Shell::find_action(const CommandSpec &command_spec,
                                   CommandSpec **matching_spec, bool *is_alias) {
-  for (size_t i = 0; i < get_actions().size(); ++i) {
-    Action *action = get_actions()[i];
+  // sort such that all "trash purge schedule ..." actions come before
+  // "trash purge"
+  std::vector<Action *> actions(get_actions());
+  std::sort(actions.begin(), actions.end(), [](auto lhs, auto rhs) {
+    return lhs->command_spec.size() > rhs->command_spec.size();
+  });
```
Comment (trociny) [inline src/tools/rbd/Shell.cc:309]: nit: alternatively it seems we could just use `std::multiset` for storing the actions, not to sort them on every `find_action` call.  But taking that `find_action` is called only once(twice) I don't mind.

**Review by trociny — APPROVED** [review]
LGTM

File: src/tools/rbd/Shell.cc (hunk 1 above)
Comment (idryomov) [reply src/tools/rbd/Shell.cc:309]: Actions are also sorted alphabetically when printing help, so I opted for keeping the main vector unsorted and sortitng on-demand with the required comparator.

**Comment by idryomov** [comment]
No actual failures:

https://pulpito.ceph.com/dis-2022-08-13_21:21:42-rbd-wip-dis-testing-1-distro-default-smithi/
https://pulpito.ceph.com/dis-2022-08-14_06:49:31-rbd-wip-dis-testing-1-distro-default-smithi/

**Pull request merged**